dist/models/message.d.ts

```diff
@@ -15,19 +15,76 @@ declare namespace Message {
         deliveryType?: DeliveryType;
         link?: string;
         type?: Types.ObjectId;
+        templateID?: string;
         richContent?: RichContent.Model;
         recipients?: Types.ObjectId[] | User.Model[];
         userGroupIDs?: Types.ObjectId[];
         individualUserIDs?: Types.ObjectId[];
+        content?: Content;
     }
     enum SubjectType {
         Post = "POST",
-        Event = "EVENT"
+        Event = "EVENT",
+        Reservation = "RESERVATION"
     }
     enum DeliveryType {
         Push = "PUSH",
         Email = "EMAIL",
         Text = "TEXT"
     }
+    type Content = Rsvp | UnRsvp | Welcome | Application | MembershipInquiry | MembershipInquiryRes | PublicRsvp | ServiceRequest | NewProviderRequest;
+    enum Type {
+        Rsvp = "Rsvp",
+        UnRsvp = "UnRsvp",
+        Welcome = "Welcome",
+        Application = "Application",
+        MembershipInquiry = "MembershipInquiry",
+        MembershipInquiryRes = "MembershipInquiryRes",
+        PublicRsvp = "PublicRsvp",
+        ServiceRequest = "ServiceRequest",
+        NewProviderRequest = "NewProviderRequest"
+    }
+    interface BaseMessageType {
+        type: Type;
+    }
+    interface Rsvp extends BaseMessageType {
+        type: Type.Rsvp | Type.UnRsvp;
+        eventID: Types.ObjectId;
+        userID: Types.ObjectId;
+        reservationID: Types.ObjectId;
+    }
+    interface UnRsvp extends Rsvp {
+    }
+    interface ServiceRequest extends BaseMessageType {
+        type: Type.ServiceRequest;
+        calendarID: Types.ObjectId;
+        eventID: Types.ObjectId;
+        userID: Types.ObjectId;
+        reservationID: Types.ObjectId;
+    }
+    interface NewProviderRequest extends BaseMessageType {
+        type: Type.NewProviderRequest;
+        calendarID: Types.ObjectId;
+    }
+    interface Welcome extends BaseMessageType {
+        type: Type.Welcome;
+        userID: Types.ObjectId;
+        password: string;
+    }
+    interface Form extends BaseMessageType {
```

Comment:
@sgruse parking all these here feels like were polluting the namespace. We've talked about it - but ok knowing we're actively refactoring. Remember to hit that though

Reply:
Will do